source/common/network/socket_impl.cc

```diff
@@ -122,5 +127,22 @@ absl::optional<Address::IpVersion> SocketImpl::ipVersion() const {
   return absl::nullopt;
 }
 
+bool SocketImpl::isSocketV6Only(const Network::Address::InstanceConstSharedPtr address) const { 
```

Comment:
Potentially this could also be moved to the address impl

Reply:
Because it depends on `io_handle_`, I'm not sure we can do the move. As per my general comment, this is needed because we "forget" what we set on the `Socket`/`IoHandle` at creation time. We could potentially improve that.